packages/docsearch-react/src/DocSearchModal.tsx

```diff
@@ -173,43 +172,36 @@ export function DocSearchModal({
             ];
           }
 
-          return getAlgoliaResults<DocSearchHit>({
-            searchClient,
-            queries: [
-              {
-                indexName,
-                query,
-                params: {
-                  attributesToRetrieve: [
-                    'hierarchy.lvl0',
-                    'hierarchy.lvl1',
-                    'hierarchy.lvl2',
-                    'hierarchy.lvl3',
-                    'hierarchy.lvl4',
-                    'hierarchy.lvl5',
-                    'hierarchy.lvl6',
-                    'content',
-                    'type',
-                    'url',
-                  ],
-                  attributesToSnippet: [
-                    `hierarchy.lvl1:${snippetLength.current}`,
-                    `hierarchy.lvl2:${snippetLength.current}`,
-                    `hierarchy.lvl3:${snippetLength.current}`,
-                    `hierarchy.lvl4:${snippetLength.current}`,
-                    `hierarchy.lvl5:${snippetLength.current}`,
-                    `hierarchy.lvl6:${snippetLength.current}`,
-                    `content:${snippetLength.current}`,
-                  ],
-                  snippetEllipsisText: '…',
-                  highlightPreTag: '<mark>',
-                  highlightPostTag: '</mark>',
-                  hitsPerPage: 20,
-                  ...searchParameters,
-                },
-              },
-            ],
-          })
+          return searchClient
```

Comment:
With the new Autocomplete Requester API, it's easier for us to use the client to create sources on the fly.